From 488507385527e2ef7c8f9624582225fb57f8a481 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Mon, 23 May 2016 23:56:46 +0200
Subject: [PATCH] libxl: enable e820_host by default if there is any PCI device
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

Follow xl behaviour. This is necessary for some PCI devices to work at
all.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 tools/libxl/libxl_create.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tools/libxl/libxl_create.c b/tools/libxl/libxl_create.c
index de5d27f..fd3be87 100644
--- a/tools/libxl/libxl_create.c
+++ b/tools/libxl/libxl_create.c
@@ -365,7 +365,6 @@ int libxl__domain_build_info_setdefault(libxl__gc *gc,
         libxl__rdm_setdefault(gc, b_info);
         break;
     case LIBXL_DOMAIN_TYPE_PV:
-        libxl_defbool_setdefault(&b_info->u.pv.e820_host, false);
         if (b_info->video_memkb == LIBXL_MEMKB_DEFAULT)
             b_info->video_memkb = 0;
         if (b_info->shadow_memkb == LIBXL_MEMKB_DEFAULT)
@@ -894,6 +893,12 @@ static void initiate_domain_create(libxl__egc *egc,
         goto error_out;
     }
 
+    /* Enable e820_host by default if there is any PCI device passed in */
+    if (d_config->c_info.type == LIBXL_DOMAIN_TYPE_PV) {
+        libxl_defbool_setdefault(&d_config->b_info.u.pv.e820_host,
+                d_config->num_pcidevs);
+    }
+
     ret = libxl__domain_create_info_setdefault(gc, &d_config->c_info);
     if (ret) {
         LOG(ERROR, "Unable to set domain create info defaults");
-- 
2.1.0

